tuplex/codegen/src/FunctionRegistry.cc

```diff
@@ -966,6 +966,301 @@ namespace tuplex {
+                BasicBlock *bb_isinf = BasicBlock::Create(builder.getContext(), "cmp_inf", builder.GetInsertBlock()->getParent());
+                BasicBlock *bb_infres = BasicBlock::Create(builder.getContext(), "opt_isinf", builder.GetInsertBlock()->getParent());
+                BasicBlock *bb_standard = BasicBlock::Create(builder.getContext(), "opt_standard", builder.GetInsertBlock()->getParent());
+                BasicBlock *bb_done = BasicBlock::Create(builder.getContext(), "cmp_done", builder.GetInsertBlock()->getParent());
+
+                // allocate space for return value
+                auto val = _env.CreateFirstBlockAlloca(builder, _env.i1Type());
+
+                // ; first block
+                // ; isnan start
+                // %5 = bitcast double %0 to i64
+                // %6 = lshr i64 %5, 32
+                // %7 = trunc i64 %6 to i32
+                // %8 = and i32 %7, 2147483647
+                // %9 = trunc i64 %5 to i32
+                // %10 = icmp ne i32 %9, 0
+                // %11 = zext i1 %10 to i32
+                // %12 = add nuw i32 %8, %11
+                // %13 = icmp ugt i32 %12, 2146435072
+                // ; isnan end
+                const std::vector<tuplex::codegen::SerializableValue> isnan_argx{SerializableValue(x, _env.i64Const(sizeof(int64_t)))};
+                auto is_x_nan = createMathIsNanCall(builder, _env.doubleType(), _env.getBooleanType(), isnan_argx);
+                // br i1 %13, label %47, label %14
+                builder.CreateStore(_env.boolConst(false), val);
+                builder.CreateCondBr(is_x_nan, bb_done, bb_nany);
+
+                // ; bb_nany
+                // 14:    ; preds = %4
+                // ; isnan start
+                // %15 = bitcast double %1 to i64
+                // %16 = lshr i64 %15, 32
+                // %17 = trunc i64 %16 to i32
+                // %18 = and i32 %17, 2147483647
+                // %19 = trunc i64 %15 to i32
+                // %20 = icmp ne i32 %19, 0
+                // %21 = zext i1 %20 to i32
+                // %22 = add nuw i32 %18, %21
+                // %23 = icmp ugt i32 %22, 2146435072
+                // ; isnan end
+                builder.SetInsertPoint(bb_nany);
+                const std::vector<tuplex::codegen::SerializableValue> isnan_argy{SerializableValue(y, _env.i64Const(sizeof(int64_t)))};
+                auto is_y_nan = createMathIsNanCall(builder, _env.doubleType(), _env.getBooleanType(), isnan_argy);
+                // br i1 %23, label %47, label %24
+                builder.CreateStore(_env.boolConst(false), val); // should overwrite value from first block
+                builder.CreateCondBr(is_y_nan, bb_done, bb_isinf);
+
+                // ; bb_isinf
+                // 24:    ; preds = %14
+                builder.SetInsertPoint(bb_isinf);
+                // %25 = fcmp oeq double %0, 0x7FF0000000000000
+                auto x_pinf = builder.CreateFCmpOEQ(x, D_PINFINITY);
```

Comment:
I think it's great to use names like D_PINFINITY here! above in the other function, magic constants where used. Def. better style to define names :)